inference-engine/src/mkldnn_plugin/utils/shape_inference/shape_inference.cpp

```diff
@@ -140,6 +141,10 @@ void shape_inference(ov::Node* op,
         shape_infer(node, input_shapes, output_shapes);
     } else if (auto node = ov::as_type<ov::opset1::OneHot>(op)) {
         shape_infer(node, input_shapes, output_shapes, constant_data);
+    } else if (auto node = ov::as_type<ov::opset4::Broadcast>(op)) {
+        shape_infer(node, input_shapes, output_shapes, constant_data);
+    } else if (auto node = ov::as_type<ov::opset1::Broadcast>(op)) {
+        shape_infer(node, input_shapes, output_shapes, constant_data);
```

Comment:
This [broadcast type check](https://github.com/openvinotoolkit/openvino/blob/6d8eaae4c13fe31d81ea7b2aa54fcec0cb161446/src/core/shape_inference/include/broadcast_shape_inference.hpp#L280-L288) is only for broadcastV3. If we ignore broadcast type and change the code to `input_shapes.size() == 2 || input_shapes.size() == 3` than V1 and V3 can use the same shape_infer and casting to BroadcastBase here instead of 2 casting to each Broadcast version. What is your suggestion?

Reply:
Let's leave it as it is. Thanks for explanation!